### PR TITLE
Major version increment.

### PR DESCRIPTION
**Pull request by lrhn**
* Is null safe.
* Removes deprecated members,
* Makes implementation classes private (they're not needed
  by end-users which just need a single instance).
* Makes all the hash object constants.

Tests run without change, so the removed functionality was *never tested*.

WDYT?

### CODE REVIEW

**Comment by lrhn** [comment]
(As usual, a major version increment of a transitive dependency of `test` is a problem. We need to update web_socket_channel to support crypto 3.0.0 before we can release crypto 3.0.0.

File: pubspec.yaml [hunk 1]
```diff
@@ -1,20 +1,38 @@
...
-  convert: ^2.2.0-nullsafety
+  convert: ^3.0.0-nullsafety
   typed_data: ^1.3.0-nullsafety
 
 dev_dependencies:
   pedantic: ^1.10.0-nullsafety
   test: ^1.16.0-nullsafety
 
 dependency_overrides:
+  analyzer:
```
Comment (kevmoo) [inline pubspec.yaml:19]: you should be able to get rid of all of these...right?

Or at least use pub overrides and not git overrides

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,18 +1,12 @@
 
-## 2.2.0-nullsafety-dev
+## 3.0.0
```
Comment (natebosch) [inline CHANGELOG.md:2]: This should still be a prerelease until the stable SDK is published.

```suggestion
## 3.0.0-nullsafety.0
```

File: pubspec.yaml (hunk 1 above)
Comment (natebosch) [reply pubspec.yaml:19]: +1 on using pub overrides

**Comment by lrhn** [comment]
Now uses pub overrides where available.

Still needs an override for non-dev dependency `convert`.
We need to make `analyzer` allow `convert` >3.0.0 before that can be removed.

**Pull request merged**